### PR TITLE
refactor(inpage-nav): upgrade component - INNO-670

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Upgrade of inpage navigation.
Twig file contains only the component, a fractal template has been created to add example and include third-part javascript.

# QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] `package.json` is up-to-date and `@ec-europa/ecl-base` is part of the dependencies
- [ ] I have given the fractal status “ready” to my component
- [ ] I have declared `@define mycomponent` in the SCSS file
- [ ] I have specified `margin: 0;` on the CSS component
- [ ] I have provided tests
- [ ] I follow the naming guidelines
- [ ] the component supports composition
- [ ] the template is fully functional
- [ ] I have filled the README.md file (at least a few lines)
- [ ] I have checked the dependencies
- [ ] there are no hardcoded strings (all content come from the context)

### CODE REVIEW

File: framework/scripts/gumshoe.min.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
+/*! gumshoe v3.5.0 | (c) 2017 Chris Ferdinandi | MIT License | http://github.com/cferdinandi/gumshoe */
```
Comment (yhuard) [inline framework/scripts/gumshoe.min.js]: Embedding external libs is not option. Please remove this file. ;)

File: framework/scripts/stickybits.min.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/**
```
Comment (yhuard) [inline framework/scripts/stickybits.min.js]: And this one

File: framework/components/ecl-navigation/ecl-navigation-inpages/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@ec-europa/ecl-navigation-inpages",
+  "author": "European Commission",
+  "license": "EUPL-1.1",
+  "version": "0.3.1",
+  "description": "ECL Inpage Navigation",
+  "main": "ecl-navigation-inpages.scss",
+  "style": "ecl-navigation-inpages.scss",
+  "dependencies": {
+    "@ec-europa/ecl-base": "^0.6.0",
```
Comment (yhuard) [inline framework/components/ecl-navigation/ecl-navigation-inpages/package.json]: The other deps (gumshoe, stickybits) should be declared here :)

File: yarn.lock [hunk 4]
```diff
@@ -443,18 +443,12 @@ async@1.5.2, async@^1.4.0, async@^1.5.0:
   version "1.5.2"
```
Comment (yhuard) [inline yarn.lock]: Strange that the lock file has been updated while the package.json has not

File: framework/_partials/_preview-navigation-inpages.twig [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    </div>
+
+    <script src="{{ '/framework/scripts/europa.js' | path }}"></script>
+    {% if _target.context._demo and _target.context._demo %}
+    <script type="text/javascript">{{ _target.context._demo.scripts | raw }}</script>
+    {% endif %}
+
+    <!-- https://github.com/dollarshaveclub/stickybits -->
+    <script src="/framework/scripts/stickybits.min.js"></script>
+    <script type="text/javascript">
```
Comment (yhuard) [inline framework/_partials/_preview-navigation-inpages.twig]: As discussed, I agree that we can abstract most of the logic here in a ECL script (and just have to call something like `ECL.scrollspy('.something');`) :)

**Comment by emeryro** [comment]
PR has been updated to better handle third part javascript.
Ready for review

**Comment by yhuard** [comment]
Let's check it

File: ecl-builder.config.js [hunk 6]
```diff
@@ -57,5 +57,19 @@ module.exports = {
       from: path.resolve(__dirname, 'framework/images'),
       to: path.resolve(__dirname, outputFolder, 'images'),
     },
+    {
+      from: path.resolve(
+        __dirname,
+        'framework/components/ecl-navigation/ecl-navigation-inpages/node_modules/gumshoejs/dist/js'
```
Comment (yhuard) [inline ecl-builder.config.js]: Do you need to import both `gumshoe.js` and `gumshoe.min.js`?

File: ecl-builder.config.js [hunk 7]
```diff
@@ -57,5 +57,19 @@ module.exports = {
...
+      ),
+      to: path.resolve(__dirname, outputFolder, 'scripts'),
+    },
+    {
+      from: path.resolve(
+        __dirname,
+        'framework/components/ecl-navigation/ecl-navigation-inpages/node_modules/stickybits/dist'
```
Comment (yhuard) [inline ecl-builder.config.js]: Same remark here, please only import what you really need :)

File: framework/_partials/_preview-navigation-inpages.twig [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+      /*stickybits(
```
Comment (yhuard) [inline framework/_partials/_preview-navigation-inpages.twig]: Why is everything commented out?

File: framework/components/ecl-navigation/ecl-navigation-inpages/package.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    "@ec-europa/ecl-expandables": "^0.6.0",
+    "stickybits": "^1.3.12",
```
Comment (yhuard) [inline framework/components/ecl-navigation/ecl-navigation-inpages/package.json]: Please use strict dependencies (safer)

File: ecl-builder.config.js (hunk 7 above)
Comment (emeryro) [inline ecl-builder.config.js]: I did try, but it disn't work.. Will try again

File: framework/_partials/_preview-navigation-inpages.twig (hunk 8 above)
Comment (emeryro) [inline framework/_partials/_preview-navigation-inpages.twig]: Forgot to remove it

**Comment by emeryro** [comment]
Modifications have been done

**Pull request merged**